src/ui/mcp/oauth/security_test.go

```
@@ -0,0 +1,361 @@
+	assert.Equal(t, fiber.StatusCreated, register(`{"client_name":"Named","redirect_uris":["http://localhost:49152/callback"],"application_type":"native","token_endpoint_auth_method":"none"}`))
+	assert.Equal(t, fiber.StatusCreated, register(`{"client_name":"Named upper","redirect_uris":["http://LocalHost:49152/callback"],"application_type":"native","token_endpoint_auth_method":"none"}`))
+	assert.Equal(t, fiber.StatusBadRequest, register(`{"client_name":"Remote","redirect_uris":["http://example.com/callback"],"application_type":"native","token_endpoint_auth_method":"none"}`))
+	assert.Equal(t, fiber.StatusBadRequest, register(`{"client_name":"Lookalike","redirect_uris":["http://localhost.example.com:49152/callback"],"application_type":"native","token_endpoint_auth_method":"none"}`))
+	assert.Equal(t, fiber.StatusBadRequest, register(`{"client_name":"No port","redirect_uris":["http://localhost/callback"],"application_type":"native","token_endpoint_auth_method":"none"}`))
+	assert.Equal(t, fiber.StatusBadRequest, register(`{"client_name":"Web","redirect_uris":["http://localhost:49152/callback"],"application_type":"web","token_endpoint_auth_method":"none"}`))
+}
+
+func TestDCRAcceptsClaudeCodeNativeLocalhostRegistration(t *testing.T) {
+	_, app := newOAuthTestServer(t, testIssuer, testResource)
+	body := `{"client_name":"Claude Code (gowa)","redirect_uris":["http://localhost:60390/callback"],"grant_types":["authorization_code","refresh_token"],"response_types":["code"],"token_endpoint_auth_method":"none","application_type":"native","scope":"mcp"}`
+	req := httptest.NewRequest("POST", "/oauth/register", strings.NewReader(body))
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)